packages/x-data-grid/src/components/cell/GridCell.tsx

```diff
@@ -326,11 +328,18 @@ const GridCell = React.forwardRef<HTMLDivElement, GridCellProps>((props, ref) =>
         border: 0,
       };
     }
-    const cellStyle = {
-      '--width': `${width}px`,
-      '--height': typeof height === 'number' ? `${height}px` : height,
-      ...styleProp,
-    } as React.CSSProperties;
+
+    const cellStyle: React.CSSProperties =
+      height !== dimensions.rowHeight
+        ? {
+            '--width': `${width}px`,
+            '--height': typeof height === 'number' ? `${height}px` : height,
+            ...styleProp,
+          }
+        : {
+            '--width': `${width}px`,
+            ...styleProp,
+          };
```

Comment:
Avoid setting `--height` unless required. `.setAttribute` is one of our highest costs.

Reply:
Do we even need `--height` here? It seems to be always set to `rowHeight`:

https://github.com/mui/mui-x/blob/4e5c0875890542709a0f159d91d4ec9427bf4899/packages/x-data-grid/src/components/GridRow.tsx#L423